src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -392,23 +450,48 @@ if this_event.type == pbs.PERIODIC:
                 power.ramp_down(poweroff_vnlist)
             if poweron_vnlist:
                 power.ramp_up(poweron_vnlist)
-            up_state = pbs.ND_FREE
         else:
             # Power on/off nodes
             if poweroff_vnlist:
                 power.power_off(poweroff_vnlist)
             if poweron_vnlist:
                 power.power_on(poweron_vnlist)
-            up_state = pbs.ND_STALE
+            ready_nodes = set()
+            if sleep_node_list:
+                ready_nodes = power.power_status(sleep_node_list)
+                # For the nodes which are up remove ND_SLEEP
+                for host in ready_nodes:
+                    for vn in nodes[host]["vnodes"]:
+                        prev_state = vn.state
+                        vn.state = prev_state & ~(pbs.ND_SLEEP)
```

Comment:
I'm a little confused here.  Aren't you just getting the power status of all the nodes in the sleep nodes list?  Couldn't some of them still be coming up?

Reply:
capmc node_status returns list of nodes based on their state. Something similar to below:
{ready : [1,2,3],
standby: [4,5],
halt: [6,7,8],
on: [9,10],
off: [11,12,13]}
Here nodes which are in ready state only are the usable by PBS. And power.power_status(sleep_node_list) will return only those which are in ready state.

Reply:
Oh ok.  The name of that function is a little confusing.  It sounds like it will return the power status of the node list passed to it.